usnistgov/oar-lib-angular
# Migrate to Jest

This checkin migrate unit test from Karma to Jest.

To test:
npm run test

Test single file:
npm run test:midasauthor header.component.spec

Debug with Jest

1.	Place “debugger” to wherever you want to set the breakpoint
2.	Run following command in a console window (the command is from Jest trouble shooting website https://jestjs.io/docs/troubleshooting) in the root folder of a project or library to debug the project or library.

node --inspect-brk ../../node_modules/.bin/jest --runInBand --config ./jest.config.js

To debug a single file:

node --inspect-brk ../../node_modules/.bin/jest –runInBand --config ./jest.config.js filename

See attached instruction for more info on migrating Karma to Jest.

[Migrating unit test to Jest.docx](https://github.com/usnistgov/oar-lib-angular/files/11223647/Migrating.unit.test.to.Jest.docx)
